daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -227,11 +227,7 @@ private[lf] object Speedy {
       /* The control is what the machine should be evaluating. If this is not
        * null, then `returnValue` must be null.
        */
-      var ctrl: SExpr,
-      /* `returnValue` contains the result once the expression in `ctrl` has
-       * been fully evaluated. If this is not null, then `ctrl` must be null.
-       */
-      var returnValue: SValue,
+      var ctrl: Either[SExpr, SValue],
```

Comment:
If we go down that route, i think we should use a custom ADT with two more appropriately named constructors than Left and Right.